packages/cli_tools/README_completion.md

```diff
@@ -147,7 +156,95 @@ autoload -Uz +X compinit && compinit
 autoload -Uz +X bashcompinit && bashcompinit
 ```
 
-### Distribution
+### Installation
+
+For end users, the generated bash script can be installed directly in their
+`~/.local/share/bash-completion/completions/`.
+
+
+## Distribution
+
+Two sub-commands are provided to help command developers to distribute the
+completion capability to their end users.
+
+### Embedding the completions scripts in the command package / executable
+
+The `completion embed` sub-command is intended for the command developer rather
+than the end users. It embeds a script in the command source code, so that
+it later can be installed by end users with `completion install`.
+
+Usage:
+
+```sh
+$ my-command completion embed --help
+Embed a command line completion script in the command source code
+
+Usage: example completion embed [arguments]
+-h, --help                  Print this usage information.
+-t, --target (mandatory)    The target tool format
+
+          [completely]      Use the `completely` tool (https://github.com/bashly-framework/completely)
+          [carapace]        Use the `carapace` tool (https://carapace.sh/)
+
+-f, --script-file           Read the script file to embed from a file instead of stdin
+-o, --output-file           The Dart file name to write ("-" for stdout)
+                            (defaults to "completion_script_<target>.dart")
+-d, --output-dir            Override the directory to write the Dart source file to
+                            (defaults to "Directory: '/Users/christer/dev/cli_tools/packages/cli_tools/lib/src'")
+```
+
+### For end-users: Installing the completion script
+
+End users run the install command:
+
+```sh
+$ my-command completion install --help
+Install a command line completion script
+
+Usage: example completion install [arguments]
+-h, --help                  Print this usage information.
+-t, --target (mandatory)    The target tool format
+
+          [completely]      Use the `completely` tool (https://github.com/bashly-framework/completely)
+          [carapace]        Use the `carapace` tool (https://carapace.sh/)
+
+-e, --exec-name             Override the name of the executable
+-d, --write-dir             Override the directory to write the script to
+```
+
+Examples:
+
+```sh
+my-command completion install -t completely
```

Comment:
In the case completely has been used to generate and embed the bash completions, the end-user doesn't actually need to know about the completely tool at all, and maybe it is confusing to the user?

I think just mentioning the shell you are installing completions for is better in that case, ie. bash not completely

Reply:
Yes, I don't see this as a blocker.